pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -1556,17 +1539,76 @@ class Timestamp(_Timestamp):
         #   check that the second argument is an int.
         #
         # - Nones for the first four (legacy) arguments indicate pydatetime
-        #   keyword arguments. year, month, and day are required. As a
-        #   shortcut, we just check that the first argument was not passed.
-        #
-        # Mixing pydatetime positional and keyword arguments is forbidden!
+        #   keyword arguments. year, month, and day are required. We just
+        #   check that no positional arguments were passed.
 
         cdef:
+            object ts_input=_no_input
             _TSObject ts
-            tzinfo_type tzobj
+            tzinfo_type tzinfo, tzobj
 
-        _date_attributes = [year, month, day, hour, minute, second,
-                            microsecond, nanosecond]
+        args_len = len(args)
+
+        # GH 30543 if pd.Timestamp already passed, return it
+        # check that only ts_input is passed
+        # checking verbosely, because cython doesn't optimize
+        # list comprehensions (as of cython 0.29.x)
+        if args_len == 1 and len(kwargs) == 0 and isinstance(args[0], _Timestamp):
+            return args[0]
```

Comment:
What if a user passes `Timestamp(Timestamp(...), tz="UTC")`? I _think_ that should localize the timestamp?